src/pages/blog-post-template-page/blogPostTemplatePage.test.tsx

```diff
@@ -0,0 +1,99 @@
+import { render, screen, waitFor } from '@testing-library/react';
+import React from 'react';
+import { MemoryRouter } from 'react-router-dom';
+import { afterEach, beforeEach, describe, expect, test, vi } from 'vitest';
+import navigationBarFactory from '../../test/factories/strapi/NavigationBarFactory';
+import FooterFactory from '../../test/factories/strapi/FooterFactory';
+import { SharedDataContext } from '../../contexts/SharedDataProvider';
+import BlogPostsFactory from '../../test/factories/strapi/BlogPostsFactory';
+import BlogPostTemplatePage from './BlogPostTemplatePage';
+
+vi.mock('react-router-dom', async () => {
+  const actual = await vi.importActual('react-router-dom');
+  return {
+    ...actual,
+    useLoaderData: vi.fn(),
+  };
+});
+
+describe('BlogPostTemplatePage', () => {
+  const mockLoaderData = new BlogPostsFactory().getMockData();
```

Comment:
This variable name needs updating as it is no longer loaderData. Apologies this is probably the result of changes I made